pkg/partials/structs.go

```diff
@@ -0,0 +1,60 @@
+package partials
+
+import (
+	"encoding/json"
+)
+
+var (
+	_ PartialPayload = (*partialJSONPayload)(nil)
```

Comment:
I like where this design is headed, but instead of a single `partials/structs.go` file, I think we should have `pkg/payload/json.go` and `pkg/payload/rest.go`